pkg/storage/stores/shipper/indexgateway/indexgatewaypb/gateway.proto

```diff
@@ -2,10 +2,46 @@ syntax = "proto3";
 
 package indexgatewaypb;
 
+import "pkg/logproto/logproto.proto";
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+
+
 service IndexGateway {
     /// QueryIndex reads the indexes required for given query & sends back the batch of rows
     /// in rpc streams
     rpc QueryIndex(QueryIndexRequest) returns (stream QueryIndexResponse);
+    /// GetChunkRef returns chunk reference that match the provided label matchers
+    rpc GetChunkRef(GetChunkRefRequest) returns (GetChunkRefResponse) {};
+    rpc LabelNamesForMetricName(LabelNamesForMetricNameRequest) returns (LabelResponse)  {};
+    rpc LabelValuesForMetricName(LabelValuesForMetricNameRequest) returns (LabelResponse) {};
+}
+
+message LabelValuesForMetricNameRequest {
+    string metric_name = 1;
+    string label_name = 2;
+    int64 from = 3 [ (gogoproto.customtype) = "github.com/prometheus/common/model.Time", (gogoproto.nullable) = false];
+    int64 through = 4 [ (gogoproto.customtype) = "github.com/prometheus/common/model.Time", (gogoproto.nullable) = false];
+    string matchers = 5;
```

Comment:
double checking: shouldn't this be a repeated matcher?

Reply:
We;re passing the stringified version.